lib/linguist/heuristics.rb

```diff
@@ -420,6 +420,14 @@ def call(data)
       end
     end
 
+    disambiguate ".sol" do |data|
+      if /^pragma solidity/.match(data)
+        Language["Solidity"]
+      elsif /M02\*$/.match(data)
```

Comment:
Well, "`as is`" will mean revising your PR after the aforementioned pull gets merged. ;) Just a heads up.